### PR TITLE
Use firecloud api defined by `WFL_FIRECLOUD_URL` 

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1172

Historically, a `workload` defines an "executor" url to be the URL of cromwell/terra to submit workflows to. While this made sense for Cromwell, Terra is sufficiently different to warrant a change from this (i.e. we don't have to fear Henry's wrath if we overload it :D ). In this change, therefore, I'm fixing the firecloud api URL to "api.firecloud.org". This can be overwritten using the environment variable "WFL_FIRECLOUD_URL" or in tests via `with-temporary-environment`.

Since the "Arrays" WDL is not available via through Dockstore in app.terra.bio, I'm disabling the arrays system tests. See https://broadinstitute.atlassian.net/browse/GH-1209 for more info.

Included here is some re-working of the "firecloud" namespace, including tweaks suggested by Tom https://github.com/broadinstitute/wfl/pull/328#discussion_r580519850

### CODE REVIEW

File: api/test/wfl/tools/fixtures.clj [hunk 1]
```diff
@@ -186,3 +187,15 @@
    datarepo/delete-dataset
    f))
 
+(defn with-temporary-environment
+  "Temporarily override the environment with the key-value mapping in `env`"
+  [new-env f]
+  (util/bracket
+   #(let [prev @env/testing] (swap! env/testing merge new-env) prev)
+   #(reset! env/testing %)
+   (fn [_] (f))))
```
Comment (rexwangcc) [inline api/test/wfl/tools/fixtures.clj:196]: nice!

**Review by rexwangcc — APPROVED** [review]
LGTM!

**Review by tbl3rd — APPROVED** [review]

File: api/src/wfl/service/firecloud.clj [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      util/response-body-json))
+
+(defn get-workflow-status-by-entity
+  "Get workflow status given a Terra submission-id and entity-name."
+  [workspace {:keys [uuid inputs] :as _item}]
+  (->> (get-submission workspace uuid)
+       :workflows
+       (filter #(= (:entity-name inputs) (get-in % [:workflowEntity :entityName])))
+       first
+       :status))
```
[the remark this answers is not in the document]
Comment (ehigham) [reply api/src/wfl/service/firecloud.clj:73]: I'm not sure - this code has really just been moved + threaded in this change. Maybe I should hoist this...

File: api/src/wfl/environment.clj [hunk 3]
```diff
@@ -51,7 +52,15 @@
...
    "WFL_TERRA_DATA_REPO_URL"
    #(-> "https://jade.datarepo-dev.broadinstitute.org/")
+   "WFL_FIRECLOUD_URL"
+   #(-> "https://api.firecloud.org/")
    "WFL_WFL_URL"
    #(-> "https://dev-wfl.gotc-dev.broadinstitute.org")})
 
-(def getenv (memoize (fn [name] (or (System/getenv name) ((defaults name))))))
+(def ^:private __getenv (memoize #(or (System/getenv %) ((defaults %)))))
+(def testing "Override the environment used by `getenv` for testing." (atom {}))
```
[the remark this answers is not in the document]
Comment (ehigham) [reply api/src/wfl/environment.clj:61]: I've added an example to the `with-temporary-environment` fixture and a threatening comment to this instead.

**Pull request merged**